controllers/btpoperator_controller.go

```diff
@@ -486,8 +491,15 @@ func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev
 	}
 	logger.Info(fmt.Sprintf("got %d module resources to apply based on %s directory", len(resourcesToApply), r.getResourcesToApplyPath()))
 
+	resolvedNamespace, _, namespaceChanged, err := r.resolveNamespace(ctx, baseSecret, &logger)
```

Comment:
Too many responsibilities for this function.
And why to return value which is ignored (always).